src/d3a/models/strategy/market_maker_strategy.py

```diff
@@ -41,3 +42,9 @@ def event_market_cycle(self):
 
     def event_activate(self):
         pass
+
+    def area_reconfigure_event(self, *args, **kwargs):
+        super().area_reconfigure_event(*args, **kwargs)
+        if key_in_dict_and_not_none(kwargs, 'grid_connected') and \
+                type(kwargs['grid_connected']) == bool:
```

Comment:
Just out of curiosity: What other type it could be?

Reply:
`grid_connected ` must always be boolean.

Reply:
That reminds me, at some point we should think about using the scenario validator for live events (or do we use it even now ? I cannot recall). Should be dealt on its own story though

Reply:
You are right, we dont validate the live events with the scenario validator. shall i add a story?

Reply:
Done: https://gridsingularity.atlassian.net/browse/D3ASIM-2961

Reply:
Thanks for adding the ticket, still, the check for boolean can be removed.

Reply:
I would rather be cautious and keep it here till the validators have been added.